tests/request_processing/erica_input/v2/test_grundsteuer_input.py

```diff
@@ -1,8 +1,174 @@
 import pytest
 from pydantic import ValidationError
 
-from erica.request_processing.erica_input.v2.grundsteuer_input import Eigentuemer, Verheiratet, SteuerId, Person
-from tests.samples.grundsteuer_sample_data import get_sample_single_person_dict
+from erica.request_processing.erica_input.v2.grundsteuer_input_eigentuemer import SteuerId, Verheiratet, Person, \
+    Eigentuemer
+from erica.request_processing.erica_input.v2.grundsteuer_input_gebaeude import Ab1949, Kernsaniert, \
+    Abbruchverpflichtung, WeitereWohnraeume, WeitereWohnraeumeDetails, Garagen, GaragenAnzahl, Gebaeude
+from tests.samples.grundsteuer_sample_data import get_sample_single_person_dict, SampleGebaeude
+
+
+class TestAb1949:
+    def test_on_camel_case_should_map_to_snake_case(self):
+        input_data = {"isAb1949": True}
+
+        result = Ab1949.parse_obj(input_data)
+
+        assert result.is_ab1949 is True
+
+    def test_foo(self):
+        gebaeude = SampleGebaeude().with_baujahr("1950").build()
+        print(gebaeude)
+
+
+class TestAbbruchverpflichtung:
+    def test_on_camel_case_should_map_to_snake_case(self):
+        input_data = {"hasAbbruchverpflichtung": True}
+
+        result = Abbruchverpflichtung.parse_obj(input_data)
+
+        assert result.has_abbruchverpflichtung is True
+
+
+class TestGaragen:
+    def test_on_camel_case_should_map_to_snake_case(self):
+        input_data = {"hasGaragen": True}
+
+        result = Garagen.parse_obj(input_data)
+
+        assert result.has_garagen is True
+
+
+class TestGaragenAnzahl:
+    def test_on_camel_case_should_map_to_snake_case(self):
+        input_data = {"anzahlGaragen": 3}
+
+        result = GaragenAnzahl.parse_obj(input_data)
+
+        assert result.anzahl_garagen == 3
+
+
+class TestKernsaniert:
+    def test_on_camel_case_should_map_to_snake_case(self):
+        input_data = {"isKernsaniert": True}
+
+        result = Kernsaniert.parse_obj(input_data)
+
+        assert result.is_kernsaniert is True
+
+
+class TestWeitereWohnraeume:
+    def test_on_camel_case_should_map_to_snake_case(self):
+        input_data = {"has_weitere_wohnraeume": True}
+
+        result = WeitereWohnraeume.parse_obj(input_data)
+
+        assert result.has_weitere_wohnraeume is True
+
+
+class TestWeitereWohnraeumeFlaeche:
+    def test_on_camel_case_should_map_to_snake_case(self):
+        input_data = {"anzahl": "2", "flaeche": "42"}
+
+        result = WeitereWohnraeumeDetails.parse_obj(input_data)
+
+        assert result.anzahl == 2
+        assert result.flaeche == 42
+
+
+class TestGebaeude:
+    def test_on_is_ab1949_but_no_baujahr_should_raise_error(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche("42").with_baujahr().build()
+
+        with pytest.raises(ValidationError):
+            Gebaeude.parse_obj(gebaeude)
```

Comment:
As you explicitly included it: Should we also test if gebaeude.baujahr.baujahr is None but gebaeude.baujahr is set?